### ISSUE TITLE
Support for new Controller AI

### ISSUE THREAD

**Issue by dwot** [dalinicus/homeassistant-acinfinity#88]
**Is your feature request related to a problem? Please describe.**
Recently acquired the new Controller AI.  I've been using the homeassistant-acinfinity addon to graph and monitor my temp, humidity and vpd in the tent.  The new controller returns API results slightly differently than the previous models.  The temperature, temperatureF, humidity on the deviceInfo object reflect the outside temperature read from the Controller AI's onboard sensor rather than the remote sensor probe located inside the grow tent.  The sensor probe data is now found in a sensors array.

"sensors": [
          {
            "sensorType": 0,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 1,
            "accessPort": 1,
            "sensorData": 6880,
            "sensorOrigin": 0
          },
          {
            "sensorType": 2,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 2,
            "accessPort": 1,
            "sensorData": 5610,
            "sensorOrigin": 0
          },
          {
            "sensorType": 3,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 1,
            "accessPort": 1,
            "sensorData": 77,
            "sensorOrigin": 0
          },
          {
            "sensorType": 4,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 1,
            "accessPort": 7,
            "sensorData": 7040,
            "sensorOrigin": 0
          },
          {
            "sensorType": 6,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 1,
            "accessPort": 7,
            "sensorData": 4700,
            "sensorOrigin": 0
          },
          {
            "sensorType": 7,
            "sensorUnit": 0,
            "sensorPrecision": 3,
            "sensorTrend": 0,
            "accessPort": 7,
            "sensorData": 134,
            "sensorOrigin": 0
          }
        ],

From what I've inferred the sensorTypes are as follows:

0 - Inside Temp
2 - Inside Humidity
3 - Inside VPD
4 - Outside Temp
6 - Outside Humidity
7 - Outside VPD

It seems a bit odd to me that they return accessPort of 7 for the onBoard (outside) sensor and 0 for the remote probe (inside) sensor, but oddities and undocumented APIs go together.

I'm not sure how deep any other differences may go as I'm typically only interested in reading the sensor data for my needs.  From what I've seen, to extend functionality to the Controller AI it would take modelling and mapping in the new sensor array data to home assistant sensors.  I'm not particularly familiar with home assistant add on development or I'd try to find my way around crafting a pull request.  Thank you again for all your time and work and your consideration of this issue.  I can understand if it's not something on any sort of immediate horizon if you don't have the new controller available, but wanted to raise the issue and provide what details I've found (I maintain a simple scraper to text file app written in go to overlay these values on my grow stream).

Thanks!

**Comment by dalinicus** [comment]
Thanks for the heads up.  I do have an AI controller on order, but haven't received it yet.  Once I do my plan was to make sure the integration was compatible.  Hopefully I can look into this issue soon.

> but oddities and undocumented APIs go together.

aint that the truth lol

**Comment by dalinicus** [comment]
Still waiting on my AI controller.  Should arrive early January.

**Comment by jordanthomas** [comment]
From what I can gather `sensorType` 0 is temp in fahrenheit and `sensorType` 1 is temp in celcius. One or the other will be present in `sensors` array, depending on account settings.

**Comment by dalinicus** [comment]
Fyi.  Just got the email that my AI controller has shipped 🥳

**Comment by Myseka** [comment]
Let me know if you need any info on AI controller, It kinda works but some things are a bit off(my humidity for one)

**Comment by gbomacfly** [comment]
> Let me know if you need any info on AI controller, It kinda works but some things are a bit off(my humidity for one)

In my case, only the external values are shown, but not the internal probe.

**Comment by Myseka** [comment]
Ahh that probably is what is showing, that makes sense.

**Comment by jeffcybulski** [comment]
All of the devices show up on the ports, but none of them can be managed. Each port shows every attribute (although I do not see Max Light Level for grow lights...).

@dalinicus Are you using Wireshark or tcpdump to reverse engineer their API? It's fortunate they've not put it behind SSL yet...

**Comment by dalinicus** [comment]
Fortunate is a weird way to describe a glaring security flaw that puts all our data at risk lol.  Luckily its mostly integers for device values and not sensitive data.

Yeah I'm using packet sniffing between the phone app and the API.  Even if they enabled TLS, I have my emulator setup with proxy certs to man-in-the-middle attack myself, so it wouldn't change much unless they take other actions to encrypt their traffic outside of TLS.

Lets hope they implement the former eventually but not the latter.

**Comment by Myseka** [comment]
Hopefully they just document it anyway, and add some non ropc oauth flows. Would just make their ecosystem more valuable.

**Comment by maziggy** [comment]
If I can do anything to help getting the new controller supported, please let me know.

**Comment by Myseka** [comment]
I assume if you can MITM yourself and get the json readouts from ai
controller that is first step, then the rest is actually just handling the
code side. It my be that he already has the json structure but still has to
code it to go to entities/devices.

**Comment by Little-Sprout** [comment]
Will be glad to help with some MITM dumps. 

I have a Full HA setup along with a Full ACI (69 Pro Ai) setup including the new CO2 module (* to control a UIS SmartOutlet with CO2 Solenoid actuator). After a chat with AC Infinity yesterday, the told me that Ai does not orchestrate any of the CO2 features. 

Previously I used ESPHome with a CO2 Sensor and then created automations in HA to cycle my SmartOutlet (* not the AC Infinity one) to deliver CO2. Something like a 2 min burst every 15 minutes gave me around 1500 ppm, all while turning down my exhaust fans. Too bad ACI is missing the boat on this scenario.

A few CO2 references:

        "co2TargetSwitch": 0,
        "co2SettingMode": 0,
        "co2HighSwitch": 0,
        "co2LowSwitch": 1,
        "co2HighValue": 0,
        "co2LowValue": 400,
        "co2TargetValue": 0,
        "co2Accuracy": 0,
        "co2FanTargetSwitch": 0,
        "co2FanSettingMode": 0,
        "co2FanHighSwitch": 0,
        "co2FanLowSwitch": 0,
        "co2FanHighValue": 2150,
        "co2FanLowValue": 0,
        "co2FanTargetValue": 0,
        "co2FanAccuracy": 1,

from --> 2025-xx-xx 11:xx:xx POST http://www.acinfinityserver.com/api/dev/getdevModeSettingList

Please let me know if you want any specific info - glad to help!

**Comment by dalinicus** [comment]
Hey y'all.  I think I got a "minimal viable code change" for this.  I've tied the new sensor array into creating new sensor devices, and they look to be populating correctly.  I haven't decided what I'm going to do with the "External" sensors on the AI controller, as they're just duplicates of the existing sensors the non-AI controllers have.  For now, I've just created the duplicate entities.

I've still got unit tests to write, some documentation, and some regression testing to do, but figured I'd get this commit out as a prerelease for anyone chomping at the bit.  If you don't mind running something "unstable", feel free to update to [1.9.0-alpha](https://github.com/dalinicus/homeassistant-acinfinity/releases/tag/1.9.0-alpha1) via HACS

Let me know what you think, and if you have any questions.

![Image](https://github.com/user-attachments/assets/42437ebe-da17-4450-8736-c66445e3828c)

![Image](https://github.com/user-attachments/assets/6af46e23-ea4c-414a-964b-198e181ec32c)

![Image](https://github.com/user-attachments/assets/020fe882-0d9c-4f63-9f1a-d71237f7b316)

![Image](https://github.com/user-attachments/assets/05439c59-2962-490e-9c17-059331590784)

**Comment by gbomacfly** [comment]
Thx for the Update:)
But I can't start the addon:

```
2025-01-28 07:29:32.893 ERROR (ImportExecutor_0) [homeassistant.loader] Unexpected exception importing platform custom_components.ac_infinity.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1272, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1304, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ac_infinity/sensor.py", line 271, in <module>
    SensorType.PROBE_TEMPERATURE: ACInfinitySensorSensorEntityDescription(
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        device_class=SensorDeviceClass.TEMPERATURE,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        get_value_fn=__get_value_fn_sensor_value_temperature,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ),
    ^
TypeError: ACInfinitySensorSensorEntityDescription.__init__() missing 1 required positional argument: 'key'
2025-01-28 07:29:32.925 ERROR (MainThread) [homeassistant.loader] Unexpected exception importing platform custom_components.ac_infinity.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1272, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1304, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ac_infinity/sensor.py", line 271, in <module>
    SensorType.PROBE_TEMPERATURE: ACInfinitySensorSensorEntityDescription(
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        device_class=SensorDeviceClass.TEMPERATURE,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        get_value_fn=__get_value_fn_sensor_value_temperature,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ),
    ^
TypeError: ACInfinitySensorSensorEntityDescription.__init__() missing 1 required positional argument: 'key'
2025-01-28 07:29:32.929 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry AC Infinity ($MYEMAIL) for ac_infinity
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1272, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1304, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/ac_infinity/sensor.py", line 271, in <module>
    SensorType.PROBE_TEMPERATURE: ACInfinitySensorSensorEntityDescription(
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        device_class=SensorDeviceClass.TEMPERATURE,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        get_value_fn=__get_value_fn_sensor_value_temperature,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ),
    ^
TypeError: ACInfinitySensorSensorEntityDescription.__init__() missing 1 required positional argument: 'key'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ac_infinity/__init__.py", line 35, in async_setup_entry
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2278, in async_forward_entry_setups
    await integration.async_get_platforms(platforms)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1185, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1185, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1185, in async_get_platforms
    import_future.result()
    ~~~~~~~~~~~~~~~~~~~~^^
  [Previous line repeated 3 more times]
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1173, in async_get_platforms
    platforms.update(self._load_platforms(platform_names))
                     ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1098, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1292, in _load_platform
    raise ImportError(
        f"Exception importing {self.pkg_path}.{platform_name}"
    ) from err
ImportError: Exception importing custom_components.ac_infinity.sensor
```

**Comment by dalinicus** [comment]
> Thx for the Update :) But I can't start the addon:
> 

@gbomacfly looks like `ACInfinitySensorSensorEntityDescription` is using the old version of `ACInfinitySensorEntityDescription` for inheritance.  Did you update via HACS or by some other method?

Try [1.9.0-alpha2](https://github.com/dalinicus/homeassistant-acinfinity/releases/tag/1.9.0-alpha2).  I reverted moving `key` and just inserted unused placeholder keys for the new sensors.

**Comment by gbomacfly** [comment]
Just using HACS. Trying the new version now :)

**Comment by gbomacfly** [comment]
It started, but the sensors said unavailable:

```
2025-01-28 16:11:33.192 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up ac_infinity platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 366, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/config/custom_components/ac_infinity/sensor.py", line 473, in async_setup_entry
    description = SENSOR_DESCRIPTIONS[sensor.sensor_type]
                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 1


2025-01-28 16:11:54.742 WARNING (MainThread) [custom_components.ac_infinity.core] Unable to refresh from data update coordinator. Retry attempt 1/2



```


P.S. How can I remove no longer existing devices? (my old 69 Pro, removed in app, but still showing in hass)

**Comment by Little-Sprout** [comment]
1.9.0-alpha2 is Looking good as far as reporting on the CO2 Sensor!

![Image](https://github.com/user-attachments/assets/a43bfbc1-adca-4f0a-9ac8-74fbf86b57b1)

**Comment by dalinicus** [comment]
Is anyone else experiencing what @gbomacfly is?  I'm unable to reproduce locally.   If not, you may need to wait until I can release the full version and do a fresh install maybe?

> P.S. How can I remove no longer existing devices?

Normally those get removed when you remove the controller from the HACS settings.  If they didn't, you'll have to delete them one by one by clicking on the sensor settings.  

![Image](https://github.com/user-attachments/assets/831e8b9c-7b7e-489e-a98a-a4f766e5d080)

**Comment by gbomacfly** [comment]
I had completely removed the Integration and the HACS-Entry and reinstalled. Still the same. But the old 69 Pro is gone :)

**Comment by maziggy** [comment]
For me it is working.

**Comment by maziggy** [comment]
Just had a closer look and it seems that the new CO2 and light entities are not shown up.

<img width="280" alt="Image" src="https://github.com/user-attachments/assets/41df28a6-8224-44ab-aadc-d36d61ec264b" />
<img width="1217" alt="Image" src="https://github.com/user-attachments/assets/34fecd5c-8989-4f48-9ffa-723984d1803f" />
<img width="975" alt="Image" src="https://github.com/user-attachments/assets/0f335251-d607-4fda-a3af-449aaf3fb297" />

**Comment by dalinicus** [comment]
> keyerror: 1

okay I just noticed that the key was 1. that's interesting.  I didn't see a sensor on my device with that device type. 

@gbomacfly @maziggy  Do you guys have any additional probes plugged into the sensor ports other than the CO2/Light sensor and the standard climate probe that came with the controller?  I have the same ones as reported by dwot

0 - Inside Temp
2 - Inside Humidity
3 - Inside VPD
4 - Outside Temp
6 - Outside Humidity
7 - Outside VPD
11 - CO2
12 - Light


I can put a dummy sensor at that index temporarily until we figure out what it is.  Either way, I should probably have a key check there in case new sensor types come out before they're supported.  This probably would have eventually got caught by my unit tests if I had time to write them lol. (my day job has been busy)

**Comment by dalinicus** [comment]
![Image](https://github.com/user-attachments/assets/ba184432-555e-422d-b271-702a8870311a)

brb, selecting the perfect chair 🤦‍♂

**Comment by gbomacfly** [comment]
Yes, here are my ports:

Port 1: Controller AI+ Sanlight Evo 5 #1
Port 2: Empty
Port 3: Controller AI+ Cloudline Pro T6
Port 4: Controller AI+ Cloudray S6 Links
Port 5: Controller AI+ Cloudray S6 Rechts
Port 6: Controller AI+ Cloudforge T3
Port 7: Empty
Port 8: Empty

**Comment by jordanthomas** [comment]
@dalinicus 

> ....... I just had a thought. if sensor 0 is Fahrenheit and sensor 1 is Celsius I'm going to throw a chair.

https://github.com/dalinicus/homeassistant-acinfinity/issues/88#issuecomment-2572172328

I tried to warn you! ;)

**Comment by dalinicus** [comment]
![Image](https://github.com/user-attachments/assets/eb84a161-7a1b-4402-b770-dfd87a3fcb1e)

**Comment by dwot** [comment]
> 
> > but oddities and undocumented APIs go together.
> 
> aint that the truth lol

Truer and truer everyday it seems

**Comment by dalinicus** [comment]
Alright, I've accounted for those using proper, sophisticated temperature units. Give [1.9.0-alpha3](https://github.com/dalinicus/homeassistant-acinfinity/releases/tag/1.9.0-alpha3) a shot.  This required a change to how I was generating the hidden entity id, so you'll probably end up with some dead sensors that need cleaned up if you tried out alpha1 or alpha2.

**Comment by maziggy** [comment]
alpha3 looks good so far.

**Comment by gbomacfly** [comment]
Looks good :)

**Comment by gbomacfly** [comment]
the probe temperature is double, localized to german and native.

<img width="354" alt="Image" src="https://github.com/user-attachments/assets/eb0d2177-9cb1-4eaa-90ff-0f09ef742f9a" />

**Comment by dalinicus** [comment]
> the probe temperature is double, 

https://github.com/dalinicus/homeassistant-acinfinity/issues/88#issuecomment-2616961039

yeah the duplication is intentional for now.  The data is available in both the new sensors and the old location that the non AI controllers used.  You can chose which one you want to use.

> localized to german and native.

I'll need help with that one.  [these new localization key/values](https://github.com/dalinicus/homeassistant-acinfinity/blob/3064d215f2079d89928f20f7c1ec38c82dcbff56/custom_components/ac_infinity/strings.json#L170C1-L193C9) will need to have their counterparts added to the [de.json translation file](https://github.com/dalinicus/homeassistant-acinfinity/blob/feature/support-ai/custom_components/ac_infinity/translations/de.json).  If someone can provide me the correct values, I'll add them in.

**Comment by gbomacfly** [comment]
I would say:
```
      "probe_temperature": {
        "name": "Sensor Temperatur"
      },
      "probe_humidity": {
        "name": "Sensor Feuchtigkeit"
      },
      "probe_vapor_pressure_deficit": {
        "name": "Sensor VPD"
      },
      "external_temperature": {
        "name": "Externe Temperatur"
      },
      "external_humidity": {
        "name": "Externe Feuchtigkeit"
      },
      "external_vapor_pressure_deficit": {
        "name": "Externer VPD"
      },
      "co2_sensor": {
        "name": "CO2"
      },
      "light_sensor": {
        "name": "Beleuchtungsstärke"
      },
```

**Comment by maziggy** [comment]
#93

**Comment by maziggy** [comment]
When I change humidity calubration for the controller, I'm always running into this error:

<img width="423" alt="Image" src="https://github.com/user-attachments/assets/45efe26f-b1eb-4537-8848-1759ac2d7a4c" />
<img width="1363" alt="Image" src="https://github.com/user-attachments/assets/847ae31d-32b3-4f69-973a-d35ef2134d37" />
<img width="1335" alt="Image" src="https://github.com/user-attachments/assets/511fa280-30f1-42d1-8de2-8a2397226a38" />

BTW: with the new controller my humidity is off about 4%, temperature is fine. Do you guys also had to adjust humidity calibration?

**Comment by dalinicus** [comment]
I did see the sensors had a calibration array.  Was hoping I could get away with implementing changing settings in a followup release, but if the existing calibration settings that were implemented for 69 controllers are erroring I'm guessing that'll need tied in too before we release.

**Comment by maziggy** [comment]
Are you also going to add the new AI stuff? If I can help in any way, just let me know.

**Comment by dalinicus** [comment]
yeah I'm more than willing to add as many features as I can.  Unfortunately time is a little tight for me right now. 

I live in the frozen tundra of Minnesota; I can't really grow in the winter... so letting me know what y'all are using and want added helps a lot.  Bonus if you can help the reverse engineering, as y'all have done so far.  

PRs are always welcome, but I'm pretty anal about my code style so I many tinker with whatever gets submitted XD 

But yeah, hopefully stuff will slow down for me here soon and I can just sit down and crank out my backlog of stuff I want to code.

**Comment by maziggy** [comment]
Just wanted to swap my 69 Pro+ with the new AI controller. Unfortunately I've learned, that there are more changes which needs to be adapted. For example, now there are three sensors for port power, min, max (as before) and current. So current port power is now controlled via it's own entity.

Think I'll put the 69 Pro+ back again until this is figured out.

**Comment by MrBrown666** [comment]
So far 1.9.0-alpha3 works good for me. But I mainly use it to display stuff on a dashboard and record data, hardly ever setting anything through the integration.
Anyway I just wanted to tell you about a change in the recent mandatory firmware update. The light sensor now only reports a percentage, 0% to 100%, instead of lux. 
From what I've seen before it maxed out at 9999lx anyway.

**Comment by dalinicus** [comment]
Oh good.  Got to love an undocumented breaking API change.  /s Thanks for the heads up y'all.

Yeah sorry.  Been dealing with the complexities of an aging parent, then my own health issues flared up earlier this month.  Hoping I'll be able to carve out some time this week to get some of this recreational coding done.  🤞🏻 

Appreciate y'all patience <3

**Comment by dalinicus** [comment]
> Just wanted to swap my 69 Pro+ with the new AI controller. Unfortunately I've learned, that there are more changes which needs to be adapted. For example, now there are three sensors for port power, min, max (as before) and current. So current port power is now controlled via it's own entity.

@maziggy  Yeah looks like the update calls for the advance controller have been moved over to a completely different endpoint.  I think the best course of action for now is to pull the controls and settings for the AI controller; just leaving the sensors.  Then bring in controls and settings a handful at a time like I did with the non-AI controller.  So much for hoping the AI controller's non-AI features would be backwards compatible. :(

> Anyway I just wanted to tell you about a change in the recent mandatory firmware update. The light sensor now only reports a percentage, 0% to 100%, instead of lux.

@MrBrown666  Thanks for the heads up.  This will be fixed next release.

I'm going to work on pulling the controls and settings for the AI controller, get the testing done for what I've got, then publish an non-alpha release so the AI controller to be used in readonly mode.  I'll make new tasks to start adding controls and settings for the AI controller.  If you got features you'd like added before others, let me know and I'll prioritize those.

**Comment by dalinicus** [comment]
@maziggy If I wanted to update the German localization string from "External Temperature" to "Controller Temperature" (and humidity and vpd) would this be correct? 
```
      "controller_temperature": {
        "name": "Regler Temperatur"
      },
      "controller_humidity": {
        "name": "Regler Luftfeuchtigkeit"
      },
      "controller_vapor_pressure_deficit": {
        "name": "Regler VPD"
      },
```

**Comment by maziggy** [comment]
The translation is correct, but I would never use it because it sounds a little bit curious ;) Just use "Controller Temperature|Luftfeuchtigkeit|VPD.

**Comment by mrb50k** [comment]
Sorry if I missed the answer to this one but has anyone solved the issue where you can only see the External Temp, Humidity, and VPD from the AI controller outside the tent?  I cant seem to locate the Inside tent Temp, Humidity, and VPD that comes from the new USB-C remote probe.

**Comment by dalinicus** [comment]
Those sensors would be on the controller itself.  The probe has a separate subdevice.  In alpha3, they would be labeled as `External Temperature`, `External Humidity`, and `External VPD`

for the full release, I've renamed them to `Controller Temperature`, `Controller Humidity`, and `Controller VPD`, and removed the redundant sensors.

If they're not showing up on the alpha3 release, can you send me a screenshot of the controller device page and check your logs for any errors?

## alpha3
![Image](https://github.com/user-attachments/assets/be36e1e0-bef1-4e47-beba-00058b9af708)

## vNext
![Image](https://github.com/user-attachments/assets/7cc8f1a6-805d-4653-81cb-ae7de009d75c)

**Comment by dalinicus** [comment]
Pushed out an official [1.9.0](https://github.com/dalinicus/homeassistant-acinfinity/releases/tag/1.9.0).  some back end ids got shuffled and redundant entities removed since alpha3, so you may have some dead controls that'll need cleaned up if you tried it out.

I'll leave this issue open for a bit longer until I can create separate github issues for the future features.

**Comment by maziggy** [comment]
#96 fixes the CO2 sensor entity.

I think we now have all metrics, but complete settings are missing.
